### PR TITLE
Remove CppObj

### PR DESCRIPTION
**Pull request by LeonMatthesKDAB**
This will move all invokables onto MyObjectQt and remove the need for
the CppObj.

### CODE REVIEW

**Comment by Be-ing** [comment]
Is this still a WIP? If not, please change the title of the PR.

File: cxx-qt-gen/src/gen_rs.rs [hunk 1]
```diff
@@ -111,6 +111,82 @@ impl RustType for QtTypes {
...
+    );
+    let parameters = &i.parameters;
+    let mutablility = if i.mutable {
+        Some(quote! { mut })
+    } else {
+        None
+    };
+
+    // TODO: invokables need to also become freestanding functions that
+    // take as input a reference to both the Rs class and the CppObject
```
Comment (ahayzen-kdab) [inline cxx-qt-gen/src/gen_rs.rs:131]: not sure which usecase this is? Can't you just use the rust/rust_mut methods from the C++ pointer? but for future branches.

Comment (ahayzen-kdab) [reply cxx-qt-gen/src/gen_rs.rs:131]: Ah lol i see it's an old comment that has moved, so yup fine for now as this code will be ripped out anyway.

**Review by ahayzen-kdab — APPROVED** [review]
Looks good, lets land this as a first step so we can move forward and parallelise tasks

**Pull request merged**